packages/flutter_tools/lib/src/migrate/migrate_compute.dart

```diff
@@ -0,0 +1,625 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/terminal.dart';
+import '../flutter_project_metadata.dart';
+import '../project.dart';
+import '../version.dart';
+import 'custom_merge.dart';
+import 'migrate_result.dart';
+import 'migrate_utils.dart';
+
+// This defines files and directories that should be skipped regardless
+// of gitignore and config settings. Using `/` as a stand-in for path separator.
+const List<String> _skippedFiles = <String>[
+  'ios/Runner.xcodeproj/project.pbxproj', // Xcode managed configs that may not merge cleanly.
+  'README.md', // changes to this shouldn't be overwritten since is is user owned.
+];
+
+const List<String> _skippedDirectories = <String>[
+  '.dart_tool', // ignore the .dart_tool generated dir
+  '.git', // ignore the git metadata
+  'lib', // Files here are always user owned and we don't want to overwrite their apps.
+  'test', // Files here are typically user owned and flutter-side changes are not relevant.
+  'assets', // Common directory for user assets.
+  'build' // Build artifacts
+];
+
+bool _skipped(String localPath, FileSystem fileSystem, {Set<String>? blacklistPrefixes}) {
+  for (final String path in _skippedFiles) {
+    if (path.replaceAll('/', fileSystem.path.separator) == localPath) {
```

Comment:
All the manual string-as-path manipulation in here is asking for trouble, particularly on Windows.  For example, `getLocalPath` could likely be replaced with a version of `path.relative`.
Another example is in `_hasDirectoryInPath` which is a `FileSystemEntity` version of this logic.
https://github.com/flutter/flutter/blob/496049263e4efd68ced664ee2c90db9c93f08571/packages/flutter_tools/lib/src/commands/ide_config.dart#L83
Everywhere you are explicitly checking `fileSystem.path.separator` is probably a chance to either switch to existing `path` package API, or to instead pass through `FileSystemEntity`s and do more fidelity checking.

Also, you don't generally need to pass around `fileSystem` if you are already passing around `FileSystemEntity` since you can do `entity.fileSystem`